public/script.js

```diff
@@ -1742,6 +1742,9 @@ async function Generate(type, automatic_trigger, force_name2) {
                 if (canFitMessages()) {
                     count_exm_add++;
                 } else {
+                    // doesn't fit, so remove
+                    examplesString.slice(examplesString.length - example.length,)
+                    count_exm_add--;
```

Comment:
Does not look right to me either. The logic in this loop was:
1. Append the example to the test variable `examplesString` to see if it fits.
2. If it fits, take note of it by incrementing `count_exm_add`, which is actually used for controlling the number of examples.
3. If it doesn't, just don't do anything, not even fix `examplesString`, because it is a throwaway variable.

So this would change the logic to "if there was at least 1 example that did not fit, remove the last example that was OK, and modify a variable which is not used anymore".

Reply:
oops, true. you can revert this one.

The actual bug I wanted to fix was that examples could be sent even when there'd be no point to them
#210

Explained in there, but short version:
```
<START>
{{user}}: I do X
{{char}}: I respond Y
```
when "I respond Y" doesn't fit the context and "I do X" does, it sends
```
[Start chat]
user: I do X
[Start chat]
[... actual chat]
```
which is useless, no actual example response.

I ended up implementing the fix in Spermack which uses Slack's Claude
the changelog was "Now it removes the example if there is not an actual example (assistant) reply"